ingest/lib/ingest-queue.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  PublishCommand,
+  PublishCommandOutput,
+  SNSClient,
+} from '@aws-sdk/client-sns';
+
+/**
+ * Queues a DSpace document for ingest using the `link` attribute. This will
+ * trigger the full asynchronous ingest process for a given DSpace document.
+ *
+ * @param uuid - UUID of the DSpace item to queue for ingest.
+ * @param ingestTopicArn - SNS Topic ARN where new documents are queued.
+ * @param region - AWS region containing the infrastructure.
+ * @returns Returns the resut of the SNS PublishCommand:
+ * https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sns/interfaces/publishcommandoutput.html
+ */
+export const queueIngestDocument = async (
```

Comment:
There's not really anything specific to ingest in this function. If you changed the argument to `message` instead of `uuid`, you could put it in an `sns-utils` function. At that point, though, there isn't really anything beyond [`publi](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SNS.html#publish-property)

Reply:
My thought here was that SNS isn't really the best thing to use for ingest and it should be updated (I'd like to create an Issue for this) and if we did fix that we could just make that change here instead of fixing everything that queues up a UUID for ingest 🤷

Reply:
> Even though `client.send()` is async?

Reply:
> Even though `client.send()` is async?

Correct. `client.send()` returns a `Promise` but, because you aren't using `await` to extract the value that the promise resolves to, and are returning that promise directly, there's no need for `async` here.